### PR TITLE
feat: add aggregation sampling and pipeline aggs

### PR DESCRIPTION
**Pull request by davidkelley**

### CODE REVIEW

File: searchlite-core/tests/aggregations.rs [hunk 1]
```diff
@@ -1525,3 +1541,264 @@ fn bucket_sort_and_avg_bucket_pipeline() {
...
+      .and_then(|agg| match agg {
+        searchlite_core::api::types::AggregationResponse::MovingAvg(val) => val.value,
+        _ => None,
+      })
+      .unwrap();
+    assert!(smooth > 0.0);
+  } else {
+    panic!("expected histogram agg");
+  }
+}
```
[the remark this answers is not in the document]
Comment (davidkelley) [reply searchlite-core/tests/aggregations.rs]: Added coverage for edge cases: bucket_script now has tests for subtraction without whitespace and rejecting near-zero division, and aggregations integration includes a gap-policy test where derivative operates on a missing buckets_path to ensure InsertZeros behavior.

File: searchlite-core/src/query/aggs/mod.rs [hunk 2]
```diff
@@ -2001,13 +2763,322 @@ fn apply_pipeline_aggs(
...
+    }
+    if ch.is_ascii_digit()
+      || ch == '.'
+      || (ch == '-'
+        && chars
+          .clone()
+          .nth(1)
+          .map(|next| next.is_ascii_digit() || next == '.')
+          .unwrap_or(false))
+    {
```
[the remark this answers is not in the document]
Comment (davidkelley) [reply searchlite-core/src/query/aggs/mod.rs]: Tokenization now tracks whether a unary minus is valid (start/after operator or `(`) before folding `-` into a number literal, so cases like `5 - 3` and `a-2` parse with the correct subtraction operator.
